runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/middleware/MutateHeaders.kt

```diff
@@ -6,16 +6,28 @@
 package aws.smithy.kotlin.runtime.http.middleware
 
 import aws.smithy.kotlin.runtime.http.*
-import aws.smithy.kotlin.runtime.http.operation.SdkHttpOperation
+import aws.smithy.kotlin.runtime.http.operation.*
+import aws.smithy.kotlin.runtime.util.InternalApi
 
 /**
  * HTTP middleware feature that allows mutation of in-flight request headers
  */
-class MutateHeaders : Feature {
+@InternalApi
+class MutateHeaders(
+    override: Map<String, String> = emptyMap(),
+    append: Map<String, String> = emptyMap(),
+    setMissing: Map<String, String> = emptyMap(),
+) : ModifyRequestMiddleware {
     private val overrides = HeadersBuilder()
     private val additional = HeadersBuilder()
     private val conditionallySet = HeadersBuilder()
 
+    init {
+        override.forEach { (key, value) -> set(key, value) }
+        append.forEach { (key, value) -> append(key, value) }
+        setMissing.forEach { (key, value) -> setIfMissing(key, value) }
+    }
```

Comment:
**Question**: What are `set`, `append`, and `setIfMissing` operating on? How do these k/v pairs get into `overrides`, `additional`, and `conditionallySet` respectively?

Reply:
`set`, `append` and `setIfMissing` are functions on this type. They could just directly modify the appropriate `HeadersBuilder` if that's preferable

Reply:
Oh I missed them in the code. I see them now. It's fine as-is.